src/defabipedia/types.py

```diff
@@ -90,6 +90,13 @@ def __init__(self, address: str, name: str, abi: str | None = None, abi_path: Pa
     def contract(self, w3: Web3):
         return w3.eth.contract(address=self.address, abi=self.abi)
 
+class SwapPools(ContractSpec):
```

Comment:
I am not sure about adding non generic types like SwapPool to the defabipedia types

Reply:
I think every pooladdress with multiple tokens in it could be added to this class, making it easier to use for identifying pools in different protocols with tokens, protocol and fee. For roles_royce (and roles_api) we will add a lot of strategies that involve swaps through certain pools, which I think shouldn't be stored in there but probably here